cmd/mesh/upgrade.go

```diff
@@ -397,33 +357,16 @@ func identicalVersions(cv []manifest.ComponentVersion) bool {
 	return true
 }
 
-// genOverlayICPS reads an ICP from filename and returns an unmarshaled and validated ICPS from the spec field of ICP.
-// It separately returns a string which represents just the overlay values in the returned ICPS.
-func genOverlayICPS(filename string, force bool) (string, *v1alpha2.IstioControlPlaneSpec, error) {
+// genOverlayICPS reads an ICPS from filename and returns the YAML of it
+func genOverlayICPS(filename string) (string, error) {
```

Comment:
Is this function still necessary? We seem to be just reading a file and returning the contents, not generating anything.

Reply:
Fixed.